tests/generic_config_updater/test_dynamic_acl.py

```diff
@@ -0,0 +1,778 @@
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+
+        for rule_name, expected_content in expected_rule_contents.items():
+            expect_acl_rule_match(duthost, rule_name, expected_content)
+
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+
+def dynamic_acl_remove_forward_rules(duthost):
+    """Remove our two forward rules from the acl table"""
+    json_patch = [
+        {
+            "op": "remove",
+            "path": "/ACL_RULE/DYNAMIC_ACL_TABLE|RULE_1",
+            "value":{}
+        },
+        {
+            "op": "remove",
+            "path": "/ACL_RULE/DYNAMIC_ACL_TABLE|RULE_2",
+            "value": { }
+        }
+    ]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+
+        expect_acl_rule_removed(duthost, "RULE_1")
+        expect_acl_rule_removed(duthost, "RULE_2")
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+def dynamic_acl_remove_table(duthost):
+    """Remove an ACL Table Type from the duthost"""
+    json_patch = [
+        {
+            "op": "remove",
+            "path": "/ACL_TABLE/DYNAMIC_ACL_TABLE",
+            "value": { }
+        }
+    ]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+def dynamic_acl_remove_nonexistant_table(duthost):
+    """Remove a nonexistent ACL Table from the duthost, verify it fails"""
+    json_patch = [
+        {
+            "op": "remove",
+            "path": "/ACL_TABLE/DYNAMIC_ACL_TABLE_BAD",
+            "value": { }
+        }
+    ]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_failure(output)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+def dynamic_acl_remove_table_type(duthost):
+    """Remove an ACL Table definition from the duthost
+    As we only have one ACL Table definition on """
+    json_patch = [
+        {
+            "op": "remove",
+            "path": "/ACL_TABLE_TYPE",
+            "value": { }
+        }
+    ]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+
+
+
+def test_gcu_acl_drop_rule_creation(rand_selected_dut, ptfadapter, setup, dynamic_acl_create_table):
+    dynamic_acl_create_drop_rule_initial(rand_selected_dut, setup)
```

Comment:
Please add some function level comments to clarify the test steps and test expectations.